pkg/bpf/collection.go

```diff
@@ -41,6 +48,135 @@ func LoadCollectionSpec(path string) (*ebpf.CollectionSpec, error) {
 	return spec, nil
 }
 
+func RemoveUnreachableTailcalls(spec *ebpf.CollectionSpec) error {
+	return removeUnreachableTailcalls(spec)
+}
+
+func removeUnreachableTailcalls(spec *ebpf.CollectionSpec) error {
+	type TailCall struct {
+		referenced bool
+		visited    bool
+		spec       *ebpf.ProgramSpec
+	}
+
+	entrypoints := make([]*ebpf.ProgramSpec, 0)
+	tailcalls := make(map[uint32]*TailCall)
+
+	for _, prog := range spec.Programs {
+		var id, slot uint32
+		// Consider any program that doesn't follow the tailcall naming convention
+		// x/y to be an entrypoint.
+		if _, err := fmt.Sscanf(prog.SectionName, "%d/%v", &id, &slot); err != nil {
```

Comment:
Follow up question: why not `%d` twice?

Also: would something break if we gave this a prefix like `static_tail_call/%d` and then got rid of the `cilium_calls_map` logic below?

Reply:
Comments on the original say its so we accept hex encoding as well (`0xffff`) for example, but I don't thing this is used outside of the policy template when not replaced. I will test if removing it breaks anything.